node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/v2/HostCapacityResponse.java

```diff
@@ -0,0 +1,161 @@
+                hosts = capacityChecker.nodesFromHostnames(hostNames);
+            } catch (IllegalArgumentException e) {
+                throw new NotFoundException(e.getMessage());
+            }
+            var failure = capacityChecker.findHostRemovalFailure(hosts);
+            if (failure.isPresent() && failure.get().failureReason.failureReasons.size() == 0) {
+                root.setBool("removalPossible", false);
+                error(root, "Removing all hosts is trivially impossible.");
+            } else {
+                if (json) hostLossPossibleToSlime(root, failure, hosts);
+                else      hostLossPossibleToText(failure, hosts);
+            }
+        } else {
+            var failurePath = capacityChecker.worstCaseHostLossLeadingToFailure();
+            if (failurePath.isPresent()) {
+                if (json) zoneFailurePathToSlime(root, failurePath.get());
+                else      zoneFailurePathToText(failurePath.get());
+            } else {
+                error(root, "Node repository contained no hosts.");
+            }
+        }
+    }
+
+    private void error(Cursor root, String errorMessage) {
+        if (json) root.setString("error", errorMessage);
+        else text.append(errorMessage);
+    }
+
+    private void hostLossPossibleToText(Optional<CapacityChecker.HostFailurePath> failure, List<Node> hostsToRemove) {
+        text.append(String.format("Attempting to remove %d hosts: ", hostsToRemove.size()));
+        CapacityChecker.AllocationHistory history = capacityChecker.allocationHistory;
+        if (failure.isEmpty()) {
+            text.append("OK\n\n");
+            text.append(history);
+            if (history.oldParents().size() != hostsToRemove.size()) {
+                long emptyHostCount = hostsToRemove.size() - history.oldParents().size();
+                text.append(String.format("\nTrivially removed %d empty host%s.", emptyHostCount, emptyHostCount > 1 ? "s" : ""));
+            }
+        } else {
+            text.append("FAILURE\n\n");
+            text.append(history).append("\n");
+            text.append(failure.get().failureReason).append("\n\n");
+       }
+    }
+
+    private void zoneFailurePathToText(CapacityChecker.HostFailurePath failurePath) {
+        text.append(String.format("Found %d hosts. Failure upon trying to remove %d hosts:\n\n",
+                capacityChecker.getHosts().size(),
+                failurePath.hostsCausingFailure.size()));
+        text.append(capacityChecker.allocationHistory).append("\n");
+        text.append(failurePath.failureReason);
+    }
+
+    private void hostLossPossibleToSlime(Cursor root, Optional<CapacityChecker.HostFailurePath> failure, List<Node> hostsToRemove) {
+        var hosts = root.setArray("hostsToRemove");
+        hostsToRemove.forEach(h -> hosts.addString(h.hostname()));
+        CapacityChecker.AllocationHistory history = capacityChecker.allocationHistory;
+        if (failure.isEmpty()) {
```

Comment:
`root.setBool("removalPossible", failure.isEmpty());`

Reply:
... Not sure how I managed to not notice that one 👍